### ISSUE TITLE
Switch to yarn instead of npm

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode-extension-vscode#150]
We should consider to switch to yarn for this module, as we did with our other modules. I am just not sure about the implications for extensions that still use NPM when installing. 

Maybe @octref and @joaomoreno could chime in.

**Comment by octref** [comment]
- Yarn Question: https://github.com/yarnpkg/yarn/issues/3322
- Yarn Behavior: https://yarnpkg.com/blog/2016/11/24/lockfiles-for-all/#libraries-vs-applications and https://yarnpkg.com/lang/en/docs/yarn-lock/#toc-check-into-source-control
- NPM Question: https://github.com/npm/npm/issues/19458
- NPM Behavior: https://docs.npmjs.com/files/package-lock.json#description

It's by design that npm/yarn only cares about top-level lock files and ignore those of dependencies. So for the `vscode` library, using which package manager only affects how `vscode` locks its install dependencies.

TLDR: As long as we have the same `dependencies` in `package.json`, downstream NPM users would resolve dependencies from there and are not affected.

**Comment by bpasero** [comment]
@octref and we probably need to keep the package.lock.json around as well right?
